app/services/AtsListService.scala

```diff
@@ -101,16 +102,18 @@ class AtsListService @Inject()(
     }
 
     // TODO: Audit Events
-    for (data <- gotData) yield {
-      data.errors match {
-        case None =>
-          sendAuditEvent(account, data)
-          storeAtsListData(data)
-        case Some(IncomingAtsError("NoAtsError")) => storeAtsListData(data)
-        case Some(_)                              => Future(data)
-      }
+    gotData flatMap {
+      case AtsSuccessResponseWithPayload(payload: AtsListData) =>
+        for {
+          _    <- sendAuditEvent(account, payload)
+          data <- storeAtsListData(payload)
+        } yield {
+          Right(data)
+        }
+      case AtsNotFoundResponse(_) => Future.successful(Left(NOT_FOUND))
+      case AtsErrorResponse(_)    => Future.successful(Left(INTERNAL_SERVER_ERROR))
```

Comment:
I think we'd still want to audit on these two cases, maybe just move it outside this function?